rsc/src/main/java/com/cloudera/livy/rsc/rpc/RpcServer.java

```diff
@@ -67,11 +68,35 @@
   private final ConcurrentMap<String, ClientInfo> pendingClients;
   private final RSCConf config;
 
-  public RpcServer(RSCConf lconf) throws IOException, InterruptedException {
+  @VisibleForTesting
+  RpcServer(RSCConf lconf) throws IOException, InterruptedException {
+    this(lconf, false);
+  }
+
+  public RpcServer(RSCConf lconf, boolean isLauncher) throws IOException, InterruptedException {
     this.config = lconf;
     this.group = new NioEventLoopGroup(
         this.config.getInt(RPC_MAX_THREADS),
         Utils.newDaemonThreadFactory("RPC-Handler-%d"));
+
+    final InetSocketAddress socketAddress;
+    if (isLauncher) {
+      if (lconf.get(LAUNCHER_ADDRESS) != null && lconf.getInt(LAUNCHER_PORT) > 0) {
+        socketAddress =
+          new InetSocketAddress(lconf.get(LAUNCHER_ADDRESS), lconf.getInt(LAUNCHER_PORT));
```

Comment:
I don't change the logic, `LAUNCHER_PORT` still used for driver to connect back to livy server.